/*
 * Aspia: Remote desktop and file transfer tool.
 * Copyright (C) 2018 Dmitry Chapyshev <dmitry@aspia.ru>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto.auth;

enum Method
{
    METHOD_UNKNOWN = 0;
    METHOD_BASIC   = 1;
}

enum SessionType
{
    SESSION_TYPE_UNKNOWN        = 0;
    SESSION_TYPE_DESKTOP_MANAGE = 1;
    SESSION_TYPE_DESKTOP_VIEW   = 2;
    SESSION_TYPE_FILE_TRANSFER  = 4;
}

enum Status
{
    STATUS_UNKNOWN       = 0;
    STATUS_SUCCESS       = 1;
    STATUS_ACCESS_DENIED = 2;
    STATUS_CANCELED      = 3;
}

message LogonRequest
{
    Method method = 1;
}

message ServerChallenge
{
    bytes nonce = 1;
}

message ClientChallenge
{
    SessionType session_type = 1;
    string username          = 2;
    bytes session_key        = 3;
}

message LogonResult
{
    Status status = 1;
}

message ClientToHost
{
    LogonRequest logon_request       = 1;
    ClientChallenge client_challenge = 2;
}

message HostToClient
{
    ServerChallenge server_challenge = 1;
    LogonResult logon_result         = 2;
}
